backend/src/GraphQl/MethodAsService/FileInputData.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+
+namespace Database.GraphQl.MethodAsService;
+
+public sealed record FileInputData(
+    Data Data
+);
+
+public sealed record Data(
+    ComponentCharacteristics ComponentCharacteristics,
+    IReadOnlyList<DataPoint> DataPoints
+);
+
+public sealed record ComponentCharacteristics(
+    string Surface,
+    string Symmetries
+);
+
+public sealed record DataPoint(
+    Incidence Incidence,
+    Emergence Emergence,
+    Results Results
+);
+
+public sealed record Incidence(
+    Wavelengths Wavelengths,
+    Direction Direction
+);
+
+public sealed record Emergence(
+    Direction Direction
+);
+
+public sealed record Results(
+    double Transmittance,
+    double? Reflectance = null,
+    ColorTransmittance? ColorTransmittance = null,
+    ColorReflectance? ColorReflectance = null
+);
+
+public sealed record Direction(
+    int Polar
+);
+
+public sealed record Wavelengths(
+    int Wavelength
```

Comment:
In general, wavelengths is more complex. https://github.com/building-envelope-data/api/blob/48fca7aef7cf813d4a63b7a8497409448fff6406/schemas/opticalData.json#L324-L335 It can be https://github.com/building-envelope-data/api/blob/48fca7aef7cf813d4a63b7a8497409448fff6406/schemas/opticalData.json#L227C6-L237 or https://github.com/building-envelope-data/api/blob/48fca7aef7cf813d4a63b7a8497409448fff6406/schemas/opticalData.json#L309-L323 Your model only encompasses a simplified version of the latter case. The `Wavelength` is a non-negative JSON number according to https://github.com/building-envelope-data/api/blob/48fca7aef7cf813d4a63b7a8497409448fff6406/schemas/number.json#L201-L205 and not just an integer. So the type here should be a float or double.